debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModel_V2.java

```diff
@@ -67,4 +69,36 @@ public UDATA getTotalFootprintInBytesWithHeader(J9IndexableObjectPointer arrayPt
 		UDATA totalFootprint = spineSize.add(externalArrayletSize);
 		return totalFootprint;
 	}
+
+	/**
+	 * Determine the validity of the data address belonging to arrayPtr.
+	 *
+	 * @param arrayPtr array object who's data address validity we are checking
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @return true if the data address of arrayPtr is valid, false otherwise
+	 */
+	@Override
+	public boolean hasCorrectDataAddrPointer(J9IndexableObjectPointer arrayPtr) throws CorruptDataException
+	{
+		boolean hasCorrectDataAddrPointer;
+		try {
+			UDATA dataSizeInBytes = getDataSizeInBytes(arrayPtr);
+			VoidPointer dataAddr = J9IndexableObjectHelper.getDataAddrForIndexable(arrayPtr);
+			boolean isValidDataAddrForOffHeapObject = isIndexableObjectOffHeap(dataAddr, dataSizeInBytes);
```

Comment:
Please remove this local and make the call to `isIndexableObjectOffHeap()` only in the `else` block where the result is needed.